code generators/de.fhdo.lemma.model_processing.code_generation.mongodb/src/main/java/de/fhdo/lemma/model_processing/code_generation/mongodb/intermediate_model_validation/MongoDBIntermediateModelValidator.xtend

```diff
@@ -0,0 +1,48 @@
+package de.fhdo.lemma.model_processing.code_generation.mongodb.intermediate_model_validation
+
+import de.fhdo.lemma.model_processing.annotations.IntermediateModelValidator
+import de.fhdo.lemma.model_processing.builtin_phases.intermediate_model_validation.AbstractIntermediateDeclarativeValidator
+import de.fhdo.lemma.operation.intermediate.IntermediatePackage
+import de.fhdo.lemma.operation.intermediate.IntermediateContainer
+import org.eclipse.xtext.validation.Check
+import de.fhdo.lemma.operation.intermediate.IntermediateInfrastructureNode
+
+/**
+ * The MongoDB intermediate model validator is responsible for checking general aspects for
+ * the MongoDB code generator.
+ *
+ * @author <a href="mailto:philip.wizenty@fh-dortmund.de">Philip Wizenty</a>
+ */
+@IntermediateModelValidator
+class MongoDBIntermediateModelValidator extends AbstractIntermediateDeclarativeValidator {
+    /**
+     * Get the namespace for the intermediate operation model
+     */
+    override getLanguageNamespace() {
+        return IntermediatePackage.eNS_URI
+    }
+
+    /**
+     * Check if the container uses a MongoDB technology model
+     */
+    @Check
+    def checkMongoDBTechnologyForContainer(IntermediateContainer container) {
+        val mongoDBTechnology = container.imports.exists[t | t.name.toLowerCase == "mongodb"]
```

Comment:
Shouldn't the qualified technology name be checked here? If so, also reuse constant value introduced above.